examples/resources/kentik-synthetic_test-bulk-create/one_agent_per_country/resource.tf

```diff
@@ -0,0 +1,53 @@
+// Create a test with list of agents - one agent per country.
+
+data "kentik-synthetics_agents" "agents" {}
+
+locals {
+  country_to_ids_map = {for agent in data.kentik-synthetics_agents.agents.items: agent.country => agent.id...}
+  agent_ids = [for key, val in local.country_to_ids_map: sort(val)[0]]
```

Comment:
That IMHO won't do it. You need to sort the list of agents you are picking from (`data.kentik-synthetics_agents.agents.items`). With that the lists of agents for each country in the map will be stable and hence picking the first one in each country produces stable selection (as long as all agents remain available).

I've also only now noticed, that you are taking 1 agent in *every* country in which Kentik has agents. That is not all that useful. Please, change it to picking 1 agent in a list of countries. You may even combine it with the "agents-filtered-by-country-test" test by adding an input parameter for max. number of agents per country. That would be closest to an actual RealWorld(tm) use case :).

Reply:
I've addressed the second part of your comment(selecting agents from given countries)


About the first part, I'm not sure if I understand the difference in sorting order.
If we sort lists of agents ids for each country, we will always get the same id first, as long as all agents are available and no new agents were added. The same problem(adding/removing agents) is present with sorting all agents first. Moreover, I'm coming to the conclusion that after we pick ids for each country, the resulting lists will be the same in both approaches:

Splitting a sorted list on arbitrary condition = Splitting a list on arbitrary condition and then sorting resulting lists.


And NGL, I would just sort the agents if I knew how. Unfortunately, the sort function(https://www.terraform.io/docs/language/functions/sort.html) accepts only strings, and the current order of things allows me to preserve the information about the country, which I would lose had I extracted the ids to sort them first.

Reply:
IMO the `sort(val)[0]` (sorting agent_ids) is correct solution.

Reply:
I agree. I've misread the HCL code as sorting the final selection. Sorting list of agents by country is really equivalent to sorting the initial list.